validator/accounts/v2/wallet/wallet.go

```diff
@@ -100,35 +100,76 @@ func New(cfg *Config) *Wallet {
 	}
 }
 
-// Exists check if a wallet at the specified directory
-// exists and has valid information in it.
-func Exists(walletDir string) error {
+// Exists checks if directory at walletDir exists
+func Exists(walletDir string) (bool, error) {
 	dirExists, err := fileutil.HasDir(walletDir)
 	if err != nil {
-		return errors.Wrap(err, "could not parse wallet directory")
+		return false, errors.Wrap(err, "could not parse wallet directory")
 	}
-	if dirExists {
-		isEmptyWallet, err := isEmptyWallet(walletDir)
-		if err != nil {
-			return errors.Wrap(err, "could not check if wallet has files")
+	return dirExists, nil
+}
+
+// IsValid checks if a folder contains a single key directory such as `derived`, `remote` or `direct`.
+// Returns true if one of those subdirectories exist, false otherwise.
+func IsValid(walletDir string) (bool, error) {
+	expanded, err := fileutil.ExpandPath(walletDir)
+	if err != nil {
+		return false, err
+	}
+	f, err := os.Open(expanded)
+	if err != nil {
+		return false, err
+	}
+	defer func() {
+		if err := f.Close(); err != nil {
+			log.Debugf("Could not close directory: %s", expanded)
 		}
-		if isEmptyWallet {
-			return ErrNoWalletFound
+	}()
+	names, err := f.Readdirnames(-1)
+	if err != nil {
+		return false, err
+	}
+
+	if len(names) == 0 {
+		return false, ErrNoWalletFound
+	}
+
+	// Count how many wallet types we have in the directory
+	numWalletTypes := 0
+	for _, name := range names {
+		// Nil error means input name is `derived`, `remote` or `direct`
+		_, err = v2keymanager.ParseKind(name)
+		if err == nil {
+			numWalletTypes++
 		}
-		return nil
 	}
-	return ErrNoWalletFound
+	if numWalletTypes == 1 {
+		return true, nil
+	}
+	return false, nil
```

Comment:
This can be simplifed to `return numWalletTypes == 1, nil`

Reply:
Addressed.  Thanks